presto-cassandra/src/main/java/com/facebook/presto/cassandra/ClusteringKeysUtils.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cassandra;
+
+import com.facebook.presto.cassandra.util.CassandraCqlUtils;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.predicate.Domain;
+import com.facebook.presto.spi.predicate.Range;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.cassandra.util.CassandraCqlUtils.toCQLCompatibleString;
+import static com.google.common.collect.Sets.cartesianProduct;
+
+public final class ClusteringKeysUtils
```

Comment:
Make this class non static. Call it `ClusteringKeyPredicatesExtractor`

Pass `CassandraTable` and `TupleDomain<ColumnHandle> unenforcedPredicates` to the constructor. 

`unenforcedPredicates` - are the predicated returned from the partition manager.

This class should have 2 methods:

- `List<String> getClusteringKeyPredicates()`
- `TupleDomain<ColumnHandle> getUnenforcedPredicates()`